third-party/crowdsec/src/test/java/org/apache/james/CrowdsecIntegrationTest.java

```diff
@@ -0,0 +1,153 @@
+
+    @RegisterExtension
+    public TestIMAPClient testIMAPClient = new TestIMAPClient();
+
+    @RegisterExtension
+    static HAProxyExtension haProxyExtension = new HAProxyExtension();
+
+    private static final String DOMAIN = "domain.tld";
+    private static final String BOB = "bob@" + DOMAIN;
+    private static final String BOB_PASSWORD = "bobPassword";
+    private static final String BAD_PASSWORD = "badPassword";
+    private static final ConditionFactory CALMLY_AWAIT = Awaitility
+        .with().pollInterval(ONE_HUNDRED_MILLISECONDS)
+        .and().pollDelay(ONE_HUNDRED_MILLISECONDS)
+        .await();
+    private static final String LOCALHOST_IP = "127.0.0.1";
+
+    private SocketChannel clientConnection;
+
+    @BeforeEach
+    void setup(GuiceJamesServer server) throws Exception {
+        server.getProbe(DataProbeImpl.class).fluent()
+            .addDomain(DOMAIN)
+            .addUser(BOB, BOB_PASSWORD);
+    }
+
+    @Test
+    void ipShouldBeBannedByCrowdSecWhenFailingToImapLoginThreeTimes(GuiceJamesServer server) {
+        // GIVEN an IP failed to log in 3 consecutive times in a short period
+        IntStream.range(0, 3)
+            .forEach(any ->
+                assertThatThrownBy(() -> testIMAPClient.connect("127.0.0.1", server.getProbe(ImapGuiceProbe.class).getImapPort())
+                    .login(BOB, BAD_PASSWORD))
+                    .isInstanceOf(IOException.class)
+                    .hasMessage("Login failed"));
+
+        // THEN connection from the IP would be blocked. CrowdSec takes time to processing the ban decision therefore the await below.
+        CALMLY_AWAIT.atMost(Durations.TEN_SECONDS)
+            .untilAsserted(() -> assertThatThrownBy(() -> testIMAPClient.connect("127.0.0.1", server.getProbe(ImapGuiceProbe.class).getImapPort()))
+                .isInstanceOf(EOFException.class)
+                .hasMessage("Connection closed without indication."));
+
+    }
+
+    @Test
+    void ipConnectViaProxyShouldBeBannedByCrowdSecWhenFailingToImapLoginThreeTimes(GuiceJamesServer server) throws IOException, InterruptedException {
+        // GIVEN an IP failed to log in 3 consecutive times in a short period
+        String CLIENT_IP = "127.0.0.1";
+        haProxyExtension.getHaproxyContainer().execInContainer(createCommandHAProxyConfig(server));
+        clientConnection = SocketChannel.open();
+
+        clientConnection.connect(new InetSocketAddress(haProxyExtension.getHaproxyContainer().getHost(), haProxyExtension.getHaproxyContainer().getMappedPort(IMAPS_PORT)));
+        IntStream.range(0, 3)
+            .forEach(any -> {
+                try {
+                    clientConnection.write(ByteBuffer.wrap(String.format("a0 LOGIN %s %s\r\n",
+                        "TCP4", CLIENT_IP, haProxyExtension.getHaproxyContainer().getHost(), 65535, haProxyExtension.getHaproxyContainer().getMappedPort(IMAPS_PORT),
+                        BOB, BAD_PASSWORD).getBytes(StandardCharsets.UTF_8)));
```

Comment:
Why are there too many string format objects while there are just 2 `%s`?

I believe we just need `BOB, BAD_PASSWORD`.